chain/chain/src/store.rs

```diff
@@ -2217,7 +2233,8 @@ impl<'a> ChainStoreUpdate<'a> {
             | DBCol::ColPeerComponent
             | DBCol::LastComponentNonce
             | DBCol::ColComponentEdges
-            | DBCol::ColBlockOrdinal => {
+            | DBCol::ColBlockOrdinal
+            | DBCol::ColStateNumParts => {
```

Comment:
`ColStateNumParts` is not unreachable as you deleting it in clear_block_data
I think you need:
```
            DBCol:: ColStateNumParts => {
                store_update.delete(col, key);
            }
```